irohad/consensus/yac/impl/yac.cpp

```diff
@@ -150,13 +172,17 @@ namespace iroha {
 
         network_->sendState(current_leader, {vote});
         cluster_order_.switchToNext();
-        if (cluster_order_.hasNext()) {
+        auto has_next = cluster_order_.hasNext();
+        lock.unlock();
```

Comment:
should not we synchronize the other accesses to `cluster_order_` for example with shared locks?
https://github.com/hyperledger/iroha/blob/6b97c3e467ca628d3a1e7f44a22b7d3420498813/irohad/consensus/yac/impl/yac.cpp#L190
https://github.com/hyperledger/iroha/blob/6b97c3e467ca628d3a1e7f44a22b7d3420498813/irohad/consensus/yac/impl/yac.cpp#L203
https://github.com/hyperledger/iroha/blob/6b97c3e467ca628d3a1e7f44a22b7d3420498813/irohad/consensus/yac/impl/yac.cpp#L286

Reply:
and `round_` btw

Reply:
They can be synchronized with shared lock as optimization, but it will introduce more complexity to code. I am not sure that it is required at the moment.
The accesses you mentioned are covered by lock in `onState` now.